packages/knip/src/typescript/resolveModuleNames.ts

```diff
@@ -147,3 +155,19 @@ export function createCustomModuleResolver(
 
   return timerify(resolveModuleNames);
 }
+
+const declarationPathRe = /^(.*)\.d(\.[^.]+)\.ts$/;
```

Comment:
This matches a lot, but I guess I'm having the same confusion as in the comment: I think you're saying "TypeScript tries to resolve `foo.ext` to `foo.d.ext.ts`", which I doubt is true? Doesn't it resolve `index` to `index.d.ts` or `anything.really` to `anything.really.d.ts`?